node/db/src/rolling/gc.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
```

Comment:
The details of the algorithm should be documented here.

Reply:
Done